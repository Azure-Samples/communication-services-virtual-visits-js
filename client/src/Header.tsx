// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import { Stack, Text, ThemeContext, createTheme } from '@fluentui/react';
import { Theme } from '@fluentui/theme';
import { WaffleMenu } from './WaffleMenu';
import { headerContainerStyles, headerTextStyles } from './styles/Header.styles';

/**
 * @interface HeaderProps The properties that Header uses.
 * @member parentid The ID of the element that will be used to contain the panel. This is required so that the panel does not cover the entire height of the screen.
 */
interface HeaderProps {
  companyName: string;
  parentid: string;
}

export const Header = (props: HeaderProps): JSX.Element => {
  const { companyName, parentid } = props;

  return (
    <ThemeContext.Consumer>
      {(theme: Theme | undefined) => {
        if (theme === undefined) {
          theme = createTheme();
        }
        return (
          <Stack styles={headerContainerStyles(theme)} verticalAlign="center" horizontal>
            <WaffleMenu parentid={parentid} />
            <Text styles={headerTextStyles(theme)}>{companyName}</Text>
          </Stack>
        );
      }}
    </ThemeContext.Consumer>
  );
};
