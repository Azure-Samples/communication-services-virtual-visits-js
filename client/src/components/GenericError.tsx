// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import { Icon, Stack, Text } from '@fluentui/react';
import { errorIconStackStyle, errorTitleStyle } from '../styles/GenericError.styles';

interface GenericErrorProps {
  statusCode: any | undefined;
}

export const GenericError = (props: GenericErrorProps): JSX.Element => {
  const { statusCode } = props;
  const error = genericErrors[statusCode] || genericErrors['default'];

  return (
    <Stack id="generic-error">
      <Text as={'h2'} className={errorIconStackStyle}>
        <Icon iconName="error" />
      </Text>
      <Text as={'h3'} className={errorTitleStyle}>
        {error.title}
      </Text>
      <Text>{error.description}</Text>
    </Stack>
  );
};

const genericErrors = {
  default: {
    title: 'Unknown error',
    description: 'An unknown error has occurred.'
  },
  404: {
    title: '404: Page not found',
    description: 'The requested page could not be found.'
  },
  500: {
    title: '500: Internal server error',
    description: 'The server has encountered an error. Refresh the page to try again.'
  }
};
