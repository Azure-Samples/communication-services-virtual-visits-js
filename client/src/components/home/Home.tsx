// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import { DefaultButton, PartialTheme, StackItem, Theme } from '@fluentui/react';
import { Stack, Text, Image } from '@fluentui/react';
import imageCalendar from '../../assets/lightCalendarSymbol.png';
import {
  btnStackStyles,
  buttonStyle,
  containerStyles,
  font16pxStyle,
  fullScreenStyles,
  innerContainer,
  lineHeight22px,
  lineHeight28px
} from '../../styles/Home.styles';
import { FrequentlyAskedQuestions } from '../FrequentlyAskedQuestions';
import { LearnMoreItem } from '../LearnMoreItem';

export interface HomeProps {
  companyName: string;
  theme?: PartialTheme | Theme;
}

export const HomeComponent = (props: HomeProps): JSX.Element => {
  return (
    <Stack styles={fullScreenStyles}>
      <Stack
        horizontalAlign="center"
        verticalAlign="start"
        styles={containerStyles(props.theme)}
        tokens={{ childrenGap: 15 }}
      >
        <Stack styles={innerContainer}>
          <Stack verticalAlign="center" horizontalAlign="center">
            <Image width="23.75rem" height="6.25rem" src={imageCalendar} alt="calendarImage"></Image>
          </Stack>
          <Stack>
            <Text styles={lineHeight28px}>Hello,</Text>
            <Text styles={lineHeight22px}>What would you like to do?</Text>
            <Stack horizontal horizontalAlign="space-evenly" styles={btnStackStyles}>
              <StackItem>
                <DefaultButton
                  text="Book an appointment"
                  styles={buttonStyle}
                  iconProps={{ iconName: 'Calendar' }}
                  onClick={() => window.location.replace('/book')}
                />
              </StackItem>
              <StackItem>
                <DefaultButton
                  text="Join from link"
                  styles={buttonStyle}
                  iconProps={{ iconName: 'Link' }}
                  onClick={() => window.location.replace('/visit')}
                />
              </StackItem>
            </Stack>
            <FrequentlyAskedQuestions />
            <Text styles={font16pxStyle}>Learn more about Azure Communication Services</Text>
            <LearnMoreItem
              headerText={'Azure Communication Services virtual appointments'}
              headerLink={'https://learn.microsoft.com/azure/communication-services/tutorials/virtual-visits'}
              description={
                'This tutorial describes concepts for virtual appointment applications. After completing this tutorial and the associated Sample Builder, you will understand common use cases that a virtual appointment delivers'
              }
            />
            <LearnMoreItem
              headerText={'Azure Communication Services Rooms (Preview)'}
              headerLink={
                'https://learn.microsoft.com/samples/azure-samples/communication-services-dotnet-quickstarts/azure-communication-services---rooms-public-preview/'
              }
              description={
                'Azure Communication Services provides a concept of a room for developers who are building structured conversations such as virtual appointments or virtual events. Rooms currently allow voice and video delivers'
              }
            />
            <LearnMoreItem
              headerText={' Get Started with UI Library'}
              headerLink={
                'https://learn.microsoft.com/azure/communication-services/quickstarts/ui-library/get-started-composites?tabs=kotlin&pivots=platform-web'
              }
              description={
                'Get Started with Azure Communication Services UI Library to quickly integrate communication experiences into your applications. In this quickstart, learn how to integrate UI Library composites into an application and set up...'
              }
            />
          </Stack>
        </Stack>
      </Stack>
    </Stack>
  );
};
