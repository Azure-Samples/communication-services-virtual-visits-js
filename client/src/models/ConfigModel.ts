// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import { Theme } from '@fluentui/theme';

export type PostCallSurveyType = 'msforms' | 'custom' | 'onequestionpoll';
export interface MSFormsSurveyOptions {
  surveyUrl: string;
}
export interface CustomSurveyOptions {
  surveyUrl: string;
}

export type OneQuestionPollType = 'likeOrDislike' | 'rating' | 'text';

export interface OneQuestionPollOptions {
  title?: string;
  prompt: string;
  pollType: OneQuestionPollType;
  answerPlaceholder?: string;
  saveButtonText: string;
}

export interface PostCallConfig {
  survey: {
    type: PostCallSurveyType;
    options: MSFormsSurveyOptions | CustomSurveyOptions | OneQuestionPollOptions;
  };
}
export interface AppConfigModel {
  communicationEndpoint: string;
  microsoftBookingsUrl: string;
  chatEnabled: boolean;
  screenShareEnabled: boolean;
  companyName: string;
  theme: Theme;
  waitingTitle: string;
  waitingSubtitle: string;
  logoUrl: string;
  postCall?: PostCallConfig;
}
