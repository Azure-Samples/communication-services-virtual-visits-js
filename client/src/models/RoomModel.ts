// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import { RoomCallLocator } from '@azure/communication-calling';

export enum RoomParticipantRole {
  presenter = 'Presenter',
  attendee = 'Attendee',
  consumer = 'Consumer'
}

export interface RoomParticipant {
  id: string;
  role: RoomParticipantRole;
}

export interface JoinRoomResponse {
  token: string;
  participant: RoomParticipant;
  invitee?: RoomParticipant;
}

export interface CreateRoomResponse {
  roomId: string;
  participants: Array<RoomParticipant>;
}

export interface RoomsInfo {
  userId: string;
  userRole: RoomParticipantRole;
  locator: RoomCallLocator;
  inviteParticipantUrl?: string;
}
