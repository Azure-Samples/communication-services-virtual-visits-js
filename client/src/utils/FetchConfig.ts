// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import { AppConfigModel } from '../models/ConfigModel';
import { generateTheme } from './ThemeGenerator';
import createError from 'http-errors';

export async function fetchConfig(): Promise<AppConfigModel | undefined> {
  const response = await fetch('/api/config', { method: 'GET' });
  if (response.status !== 200) {
    throw new createError(
      response.status,
      `Unable to retrieve config. Status: ${response.status}. Message: ${response.statusText}.`
    );
  }

  const responseContent = await response.text();
  const config = JSON.parse(responseContent);
  if (!config) {
    throw new Error(`Received empty config response.`);
  }

  const appConfigModel: AppConfigModel = {
    ...config,
    theme: generateTheme(config.colorPalette)
  };

  return appConfigModel;
}
