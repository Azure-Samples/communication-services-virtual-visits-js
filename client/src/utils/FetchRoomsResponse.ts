// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import createError from 'http-errors';
import { CreateRoomResponse, JoinRoomResponse } from '../models/RoomModel';

export const fetchRoomsResponse = async (roomId: string, userId: string): Promise<JoinRoomResponse | undefined> => {
  const response = await fetch('/api/rooms/token', {
    method: 'POST',
    body: JSON.stringify({
      roomId: roomId,
      userId: userId
    }),
    headers: {
      'content-type': 'application/json;charset=UTF-8'
    }
  });

  if (response.status !== 200) {
    throw new createError(
      response.status,
      `Unable to retrieve user token. Status: ${response.status}. Message: ${response.statusText}.`
    );
  }

  const responseContent = await response.text();
  const roomResponse = JSON.parse(responseContent);
  return roomResponse;
};

export const createRoom = async (): Promise<CreateRoomResponse> => {
  const response = await fetch('/api/rooms', {
    method: 'POST',
    headers: {
      'content-type': 'application/json;charset=UTF-8'
    }
  });
  if (response.status !== 201) {
    throw new createError(
      response.status,
      `Unable to create room. Status: ${response.status}. Message: ${response.statusText}.`
    );
  }
  const responseContent = await response.text();
  const roomsResponse = JSON.parse(responseContent);
  return roomsResponse;
};
