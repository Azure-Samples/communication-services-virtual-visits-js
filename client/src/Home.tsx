// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import { Spinner, Stack, ThemeProvider } from '@fluentui/react';
import { backgroundStyles, fullSizeStyles } from './styles/Common.styles';
import { useEffect, useState } from 'react';
import { fetchConfig } from './utils/FetchConfig';
import { GenericError } from './components/GenericError';
import { AppConfigModel } from './models/ConfigModel';
import HomeComponent from './components/home/HomeComponent';
import { Header } from './Header';
import './styles/Common.css';
import GenericContainer from './components/GenericContainer';

const PARENT_ID = 'HomeSection';

export const Home = (): JSX.Element => {
  const [config, setConfig] = useState<AppConfigModel | undefined>(undefined);
  const [error, setError] = useState<any | undefined>(undefined);

  useEffect(() => {
    const fetchData = async (): Promise<void> => {
      try {
        const config = await fetchConfig();
        setConfig(config);
      } catch (error) {
        setError(error);
      }
    };

    fetchData();
  }, []);

  if (error) {
    return <GenericError statusCode={error.statusCode} />;
  }

  if (!config) {
    // config and token not ready yet - show spinning/loading animation
    return <Spinner styles={fullSizeStyles} />;
  }

  return (
    <ThemeProvider theme={config.theme} style={{ height: '100%' }}>
      <Stack styles={backgroundStyles(config.theme)}>
        <Header companyName={config.companyName} parentid={PARENT_ID} />
        <GenericContainer layerHostId={PARENT_ID} theme={config.theme}>
          <HomeComponent
            companyName={config.companyName}
            theme={config.theme}
            onDisplayError={(error) => setError(error)}
          />
        </GenericContainer>
      </Stack>
    </ThemeProvider>
  );
};
