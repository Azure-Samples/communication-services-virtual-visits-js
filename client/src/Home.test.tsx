// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import { mount } from 'enzyme';
import * as FetchConfig from './utils/FetchConfig';
import * as CreateRoom from './utils/FetchRoomsResponse';
import { runFakeTimers } from './utils/TestUtils';
import { Header } from './Header';
import { Spinner } from '@fluentui/react';
import { Home } from './Home';
import { AppConfigModel } from './models/ConfigModel';
import { GenericError } from './components/GenericError';
import { CreateRoomResponse } from './models/RoomModel';

describe('Home', () => {
  it('should render loading spinner when config is not loaded', async () => {
    const fetchConfigSpy = jest.spyOn(FetchConfig, 'fetchConfig');
    fetchConfigSpy.mockReturnValue(Promise.resolve(undefined));

    const home = mount(<Home />);

    await runFakeTimers();

    home.update();

    const spinners = home.find(Spinner);
    const headers = home.find(Header);

    expect(spinners.length).toBe(1);
    expect(headers.length).toBe(0);
  });

  it('renders a generic error when config throws an error', async () => {
    const fetchConfigSpy = jest.spyOn(FetchConfig, 'fetchConfig');
    fetchConfigSpy.mockImplementation(
      async (): Promise<AppConfigModel | undefined> => {
        throw new Error('test error');
      }
    );

    const home = mount(<Home />);

    await runFakeTimers();

    home.update();

    const spinners = home.find(Spinner);
    const genericErrors = home.find(GenericError);

    expect(spinners.length).toBe(0);
    expect(genericErrors.length).toBe(1);
  });

  it('renders a generic error when createRoom throws an error', async () => {
    const createRoomSpy = jest.spyOn(CreateRoom, 'createRoom');
    createRoomSpy.mockImplementation(
      async (): Promise<CreateRoomResponse> => {
        throw new Error('test error');
      }
    );

    const home = mount(<Home />);

    await runFakeTimers();

    home.update();

    const spinners = home.find(Spinner);
    const genericErrors = home.find(GenericError);

    expect(spinners.length).toBe(0);
    expect(genericErrors.length).toBe(1);
  });

  it('renders a generic error when createRoom throws an error', async () => {
    const createRoomSpy = jest.spyOn(CreateRoom, 'createRoom');
    createRoomSpy.mockImplementation(
      async (): Promise<CreateRoomResponse> => {
        throw new Error('test error');
      }
    );

    const home = mount(<Home />);

    await runFakeTimers();

    home.update();

    const spinners = home.find(Spinner);
    const genericErrors = home.find(GenericError);

    expect(spinners.length).toBe(0);
    expect(genericErrors.length).toBe(1);
  });
});
