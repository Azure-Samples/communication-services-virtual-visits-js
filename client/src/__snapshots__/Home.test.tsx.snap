// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage tests should render home page 1`] = `
<div
  className="ms-Stack css-110"
>
  <div
    className="ms-Stack css-111"
  >
    <div
      className="ms-Stack css-112"
    >
      <div
        className="ms-Stack css-113"
      >
        <div
          className="ms-Image root-114"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        >
          <img
            alt="calendarImage"
            className="ms-Image-image ms-Image-image--contain ms-Image-image--portrait is-notLoaded is-fadeIn image-115"
            onError={[Function]}
            onLoad={[Function]}
          />
        </div>
      </div>
      <div
        className="ms-Stack css-116"
      >
        <span
          className="css-117"
        >
          Hello,
        </span>
        <span
          className="css-118"
        >
          What would you like to do?
        </span>
        <div
          className="ms-Stack css-120"
        >
          <div
            className="ms-Stack-inner css-119"
          >
            <div
              className="ms-StackItem css-121"
            >
              <button
                className="ms-Button ms-Button--default root-122"
                data-is-focusable={true}
                onClick={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
                type="button"
              >
                <span
                  className="ms-Button-flexContainer flexContainer-123"
                  data-automationid="splitbuttonprimary"
                >
                  <i
                    aria-hidden={true}
                    className="ms-Button-icon icon-130"
                    data-icon-name="Calendar"
                  />
                  <span
                    className="ms-Button-textContainer textContainer-124"
                  >
                    <span
                      className="ms-Button-label label-126"
                      id="id__0"
                    >
                      Book an appointment
                    </span>
                  </span>
                </span>
              </button>
            </div>
            <div
              className="ms-StackItem css-121"
            >
              <button
                aria-controls={null}
                aria-expanded={false}
                aria-haspopup={true}
                aria-roledescription="split button"
                className="ms-Button ms-Button--default ms-Button--hasMenu root-122"
                data-is-focusable={true}
                onClick={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
                type="button"
              >
                <span
                  className="ms-Button-flexContainer flexContainer-123"
                  data-automationid="splitbuttonprimary"
                >
                  <i
                    aria-hidden={true}
                    className="ms-Button-icon icon-130"
                    data-icon-name="Calendar"
                  />
                  <span
                    className="ms-Button-textContainer textContainer-124"
                  >
                    <span
                      className="ms-Button-label label-126"
                      id="id__3"
                    >
                      Start a call
                    </span>
                  </span>
                  <i
                    aria-hidden={true}
                    className="ms-Icon root-105 ms-Button-menuIcon menuIcon-127"
                    data-icon-name="ChevronDown"
                    style={
                      Object {
                        "fontFamily": undefined,
                      }
                    }
                  />
                </span>
              </button>
            </div>
            <div
              className="ms-StackItem css-121"
            >
              <button
                className="ms-Button ms-Button--default root-122"
                data-is-focusable={true}
                onClick={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
                type="button"
              >
                <span
                  className="ms-Button-flexContainer flexContainer-123"
                  data-automationid="splitbuttonprimary"
                >
                  <i
                    aria-hidden={true}
                    className="ms-Button-icon icon-130"
                    data-icon-name="Link"
                  />
                  <span
                    className="ms-Button-textContainer textContainer-124"
                  >
                    <span
                      className="ms-Button-label label-126"
                      id="id__6"
                    >
                      Join from link
                    </span>
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
        <span
          className="css-131"
        >
          Frequently asked questions
        </span>
        <div
          className="ms-Stack css-132"
        >
          <a
            className="ms-Link root-133"
            href="https://aka.ms/virtual-appointments-sample-bookings"
            onClick={[Function]}
            tabIndex={0}
            target="_blank"
          >
            <div
              className="ms-Stack css-134"
            >
              <div
                className="ms-StackItem css-135"
                style={
                  Object {
                    "fontSize": ".9375rem",
                    "fontWeight": "400",
                    "letterSpacing": "-0.015rem",
                    "lineHeight": "1rem",
                    "paddingRight": ".5rem",
                    "textDecoration": "underline",
                    "verticalAlign": "bottom",
                  }
                }
              >
                How do I change my Microsoft Bookings page URL?
              </div>
              <div
                className="ms-StackItem css-135"
                style={
                  Object {
                    "textDecoration": "none",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                    }
                  }
                >
                  <i
                    aria-hidden={true}
                    className="root-105"
                    data-icon-name="OpenInNewWindow"
                  />
                </span>
              </div>
            </div>
          </a>
        </div>
        <span
          className="css-131"
        >
          Learn more about Azure Communication Services
        </span>
        <a
          className="ms-Link root-133"
          href="https://learn.microsoft.com/azure/communication-services/tutorials/virtual-visits"
          onClick={[Function]}
          target="_blank"
        >
          <div
            className="ms-Stack css-134"
          >
            <div
              className="ms-StackItem css-135"
              style={
                Object {
                  "fontSize": ".75rem",
                  "fontWeight": "600",
                  "lineHeight": "1rem",
                  "paddingRight": ".5rem",
                  "verticalAlign": "bottom",
                }
              }
            >
              Azure Communication Services virtual appointments
            </div>
            <div
              className="ms-StackItem css-135"
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <span
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                  }
                }
              >
                <i
                  aria-hidden={true}
                  className="root-105"
                  data-icon-name="OpenInNewWindow"
                />
              </span>
            </div>
          </div>
        </a>
        <span
          className="css-136"
        >
          This tutorial describes concepts for virtual appointment applications. After completing this tutorial and the associated Sample Builder, you will understand common use cases that a virtual appointment delivers...
        </span>
        <a
          className="ms-Link root-133"
          href="https://learn.microsoft.com/en-us/azure/communication-services/concepts/rooms/room-concept"
          onClick={[Function]}
          target="_blank"
        >
          <div
            className="ms-Stack css-134"
          >
            <div
              className="ms-StackItem css-135"
              style={
                Object {
                  "fontSize": ".75rem",
                  "fontWeight": "600",
                  "lineHeight": "1rem",
                  "paddingRight": ".5rem",
                  "verticalAlign": "bottom",
                }
              }
            >
              Azure Communication Services Rooms (Preview)
            </div>
            <div
              className="ms-StackItem css-135"
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <span
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                  }
                }
              >
                <i
                  aria-hidden={true}
                  className="root-105"
                  data-icon-name="OpenInNewWindow"
                />
              </span>
            </div>
          </div>
        </a>
        <span
          className="css-136"
        >
          Azure Communication Services provides a concept of a room for developers who are building structured conversations such as virtual appointments or virtual events. Rooms currently allow voice and video calling...
        </span>
        <a
          className="ms-Link root-133"
          href="https://learn.microsoft.com/azure/communication-services/quickstarts/ui-library/get-started-composites?tabs=kotlin&pivots=platform-web"
          onClick={[Function]}
          target="_blank"
        >
          <div
            className="ms-Stack css-134"
          >
            <div
              className="ms-StackItem css-135"
              style={
                Object {
                  "fontSize": ".75rem",
                  "fontWeight": "600",
                  "lineHeight": "1rem",
                  "paddingRight": ".5rem",
                  "verticalAlign": "bottom",
                }
              }
            >
              Get Started with UI Library
            </div>
            <div
              className="ms-StackItem css-135"
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <span
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                  }
                }
              >
                <i
                  aria-hidden={true}
                  className="root-105"
                  data-icon-name="OpenInNewWindow"
                />
              </span>
            </div>
          </div>
        </a>
        <span
          className="css-136"
        >
          Get Started with Azure Communication Services UI Library to quickly integrate communication experiences into your applications. In this quickstart, learn how to integrate UI Library composites into an application and set up...
        </span>
      </div>
    </div>
  </div>
</div>
`;
